### PR TITLE
[Xamarin.Android.Build.Tasks] remove Xamarin.Android.Windows.targets

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Fixes: http://work.devdiv.io/707557

In 539954c, I was aiming to remove the
`_RegisterAndroidFilesWithFileWrites` MSBuild target (runs only on
Windows). I was having trouble getting everything to work, so I undid
this change to make the PR smaller.

Reasons to remove `_RegisterAndroidFilesWithFileWrites`:

- It looks like it *may* be a source of incremental build bugs. It
  causes `$(CleanFile)` to grow on incremental builds indefinitely...
- We should instead use the `FileWrites` item group appropriately.
- We can avoid ~81ms of build time.
```
81 ms  _RegisterAndroidFilesWithFileWrites        1 calls
```
Revisiting it now, it looks like we can remove
`Xamarin.Android.Windows.targets` completely!

- `_RegisterAndroidFilesWithFileWrites` is the only target.
- The `$(_IsRunningXBuild)` property is only used in this file.
- I moved the `$(Debugger)` property to
  `Xamarin.Android.Common.props`. I believe this value is only used in
  VS Windows, but it make more sense to be here.

## CopyGeneratedJavaResourceClasses ##

The `<CopyGeneratedJavaResourceClasses/>` MSBuild task was copying an
`R.java` file from `%TEMP%` to:

    $(IntermediateOutputPath)android\unnamedproject\unnamedproject\R.java

We only use this file in one place, since the rest of the build uses:

    $(IntermediateOutputPath)android\src\unnamedproject\unnamedproject\R.java

We can avoid copying this extra file and adding it to `FileWrites`:

- `<CopyGeneratedJavaResourceClasses/>` now optionally uses the
`DestinationTopDirectory` property
- If `DestinationTopDirectory` is blank, the original source file is
returned as the `PrimaryJavaResgenFile` output property.

We now use the `R.java` from `%TEMP%` directly, and avoid copying a
file to `$(IntermediateOutputPath)`.

## Other Changes ##

I modified the `CheckNothingIsDeletedByIncrementalClean` test to
verify that `$(CleanFile)` isn't growing on incremental builds.

Anything added to `FileWrites` during the `_CompileDex` MSBuild target
(or its dependent targets) wasn't showing up in `$(CleanFile)`! I
adjusted its `BeforeTargets` to match what I did in 539954c for
`_PrepareAssemblies`.

Two files needed to be added to `FileWrites`:

- `$(_PackagedResources)`
- `$(_GeneratedPrimaryJavaResgenFile)`

### CODE REVIEW

**Comment by jonathanpeppers** [comment]
Need to look into test failures.

**Comment by jonathanpeppers** [comment]
@dellis1972  I need to ask you about this after the break: https://github.com/xamarin/xamarin-android/blob/46f02511f8c91b133fb09fdcad2fa8fe9f2cc730/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L1795-L1801

This task creates `$(IntermediateOutputPath)android\unnamedproject\unnamedproject\R.java`, but I normally expect this file in `$(IntermediateOutputPath)android\src\unnamedproject\unnamedproject\R.java`.

Is this path wrong, or are we deliberately keeping two copies? I'm seeing the one _not_ in `src` getting deleted by `IncrementalClean`.

**Comment by jonathanpeppers** [comment]
@dellis1972 it looks like the only thing using this file: `$(IntermediateOutputPath)android\unnamedproject\unnamedproject\R.java`

Is the `<GenerateResourceDesigner/>` task right below it using `$(_GeneratedPrimaryJavaResgenFile)`:
https://github.com/xamarin/xamarin-android/blob/d1a307097419aa80608ee3636bc9f30aa6860f75/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L1263-L1277

It looks like:
- `<Aapt/>` puts `R.java` in `%TEMP%`
- `<CopyGeneratedJavaResourceClasses/>` finds it in `%TEMP%` and copies to `$(IntermediateOutputPath)android\unnamedproject\unnamedproject\R.java`
- `<GenerateResourceDesigner/>` uses `$(IntermediateOutputPath)android\unnamedproject\unnamedproject\R.java`
- `<RemoveDirFixed/>` deletes the temp dir

I'm thinking I could just rework this to use the `R.java` directly from `%TEMP%`? And not copy anything?

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AndroidUpdateResourcesTest.cs [hunk 1]
```diff
@@ -117,13 +107,13 @@ public void DesignTimeBuild ([Values(false, true)] bool isRelease, [Values (fals
 					if (useManagedParser) {
 						FileAssert.Exists (designTimeDesigner, $"{designTimeDesigner} should not have been deleted.");
 					}
-					items.Clear ();
+					var items = new List<string> ();
 					if (!useManagedParser) {
-						foreach (var file in Directory.EnumerateFiles (Path.Combine (intermediateOutputPath, "android"), "R.java", SearchOption.AllDirectories)) {
+						foreach (var file in Directory.EnumerateFiles (Path.Combine (intermediateOutputPath, "android", "src"), "R.java", SearchOption.AllDirectories)) {
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AndroidUpdateResourcesTest.cs]: @garuma we discovered we had some duplicate copies of `R.java` in:
- `obj/Debug/android/com/mypackage/R.java`
- `obj/Debug/android/src/com/mypackage/R.java`

This PR is removing the non-`src` one. I wanted to check and make sure the designer doesn't use it?

Comment (garuma) [reply src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AndroidUpdateResourcesTest.cs]: @jonathanpeppers that should be fine, we only use the compiled version and the resulting Resource.designer.cs

Thanks for checking!

**Pull request merged**